kahuna/public/js/leases/leases.js

```diff
@@ -31,60 +47,81 @@ leases.controller(
                 $window.alert('Please select an access type (Allow or Deny)');
             } else {
                 ctrl.adding = true;
-                ctrl.newLease.mediaId = ctrl.image.data.id;
                 ctrl.newLease.createdAt = new Date();
                 ctrl.newLease.access = ctrl.access;
 
-                leaseService.add(ctrl.image, ctrl.newLease)
-                    .then(() => {
-                        ctrl.updateLeases(ctrl.image);
-                    })
+                leaseService.batchAdd(ctrl.newLease, ctrl.leases, ctrl.images)
                     .catch(() =>
-                        alertFailed('Something went wrong when saving, please try again!')
+                        alertFailed('Something went wrong when saving, please try again.')
                     )
                     .finally(() => {
-                        ctrl.editing = false;
-                        ctrl.adding = false;
                         ctrl.resetLeaseForm();
                 });
             }
         };
 
+        function addLease(newLease) {
+            leaseService.batchAdd(newLease, ctrl.leases, ctrl.images);
+        }
+
+        // These events allow this control to work as a hybrid on the upload page
+        const batchAddLeasesEvent = 'events:batch-apply:add-labels';
+        const batchRemoveLeasesEvent = 'events:batch-apply:remove-labels';
+
+        if (Boolean(ctrl.withBatch)) {
+            $scope.$on(batchAddLeasesEvent,
+                    (e, leases) => leases.map(lease => addLease(lease)));
+            $scope.$on(batchRemoveLeasesEvent,
+                    () => console.log("batch-remove-leases"));
```

Comment:
We need to add a remove all leases for an image method to the lease service I think.